# Analyze data to help developer understand, type of apps users are more likely to interact
Mostly app revenue depends on ads users see on apps.
> Number of users ~ engage with apps - ads ~ More revenue

## Opening and Exploring Data

To easy explore, will create this explore_data function

In [5]:
def explore_data(dataset, start, end, row_and_column=False):
    for i in dataset[start:end]:
        print(i)
        print('\n')
    if(row_and_column):
        print("Row: ", len(dataset))
        print("Column: ", len(dataset[0]))
    

In [6]:
def getfile(filename,header = True):
    open_file  = open(filename, encoding='utf-8')
    from csv import reader
    read_file  =  reader(open_file)
    apps_data = list(read_file)
    return apps_data

In [7]:
ios = getfile('AppleStore.csv',True)
ios_header = ios[0]
ios_data = ios[1:]
print(explore_data(ios_data,0,2,row_and_column = True))
print(ios_header)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


Row:  7197
Column:  16
None
['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


The apple appstore have 7197 entries(rows) , and 16 attributes(columns)

In [9]:
android = getfile('googleplaystore.csv',True)
android_header = android[0]
android_data = android[1:]
print(explore_data(android_data,0,2,True))
print(android_header)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


Row:  10841
Column:  13
None
['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


Google Play Store data have 10841 rows and 13 columns 

 ## Cleaning Data

In [12]:
# An easy way to check if all the rows have same number of elements:

In [13]:
for i in android_data:
    if(len(i) != len(android_header)):
        print(i)
        print(android_data.index(i))

print(len(android_header))
print(len(android_data[10472]))

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']
10472
13
12


In [14]:
# We can see this row :- index 10472 have some isse
# this row have less datapoints
# we'll delete this row for this project, since we have enough other data points

In [15]:
del android_data[10472]

In [16]:
for i in android_data:
    if(len(i) != len(android_header)):
        print(i)
        print(android_data.index(i))

In [17]:
print(len(android_data))

10840


In [18]:
# We dont have an output this time , means the data has been removed also, the row count has been reduced

In [19]:
for i in ios_data:
    if(len(i) != len(ios_header)):
        print(i)
        print(ios_data.index(i))

## Deleting Duplicate rows

First to check if we have duplicate rows

In [22]:
%%time
unique_data = []
duplicate_items = []
for i in android_data:
    if i[0] in unique_data:
        duplicate_items.append(i[0])
    else:
        unique_data.append(i[0])
print('Number of duplicate items: ', len(duplicate_items))
print('Few Duplicate Items are : ', duplicate_items[:10])

Number of duplicate items:  1181
Few Duplicate Items are :  ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack']
CPU times: total: 531 ms
Wall time: 538 ms


Taking any random duplicate item to see if they are exact same or any difference is present between duplicates of same item

In [24]:
print(len(duplicate_items),'\n',len(unique_data), sep ="")

1181
9659


In [25]:
for i in android_data:
    if i[0] == 'Google Ads':
        print(i)

['Google Ads', 'BUSINESS', '4.3', '29313', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']
['Google Ads', 'BUSINESS', '4.3', '29313', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']
['Google Ads', 'BUSINESS', '4.3', '29331', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']


We can see the 4th column , i.e Reviews are different
We can't randomly choose 1, so we'll keep here the entry with highest review

In [27]:
%%time
max_review = {}
for i in android_data:
    if(i[0] in max_review and i[3]>max_review[i[0]]):
        max_review[i[0]] = i[3]
    elif(i[0] not in max_review):
        max_review[i[0]] = i[3]

print(len(max_review))
        
        

9659
CPU times: total: 15.6 ms
Wall time: 7.01 ms


In [28]:
# This max review contains all apps and their max review. We are going to consider this max review for duplicate items

In [29]:
android_clean = []
added_apps = []
for i in android_data:
    if(max_review[i[0]] == i[3] and i[0] not in added_apps):
        android_clean.append(i)
        added_apps.append(i[0])

In [30]:
print(len(android_clean))

9659


In [31]:
for i in android_clean:
    if i[0] == 'Google Ads':
        print(i)

['Google Ads', 'BUSINESS', '4.3', '29331', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']


We can see the data is cleaned (without duplicate) 

## Remove non-english apps

All the english letter have an ASCII value of less than 127.

In [35]:
def english(string):
    for i in string:
        if(ord(i)>127):
            print("Non English")
            return
    print('english')

In [36]:
english('abcdAA')
english('हिंदी टाइपिंग')
english('abcde@@@@🤗')

english
Non English
Non English


However, english apps can have emojis as well, which is having a higher ASCII  value

In [38]:
# So, we'll consider if an appname have more than 3 such characters , we'll reject that app

In [39]:
def new_english(string):
    counter = 0
    for i in string:
        if ord(i)>127:
            counter+=1
    if counter>3:
        return False
    else:
        return True

In [40]:
print(new_english('abcdAA'))
print(new_english('हिंदी टाइपिंग'))
print(new_english('abcde@@@@🤗'))

True
False
True


In [41]:
# Implementing this logic in our analysis to just extract and keep English apps

In [42]:
android_english = []
ios_english = []
for i in android_clean:
    if new_english(i[0]):
        android_english.append(i)
for x in ios_data:
    if new_english(x[1]):
        ios_english.append(x)

In [43]:
explore_data(android_english,0,3,True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Row:  9614
Column:  13


In [44]:
explore_data(ios_english,0,3,True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Row:  6183
Column:  16


We are now left with 9614 elements of android data  and 6183 rows of ios data

## Consider only Free apps

We'll only consider the apps with Price 0, i.e free apps. 

In [48]:
type(android_english[1][7])

str

In [49]:
type(ios_english[1][4])

str

In [50]:
ios_english[1][4]

'0.0'

In [51]:
# So, price :  0 , is in str format

In [52]:
android_final = []
ios_final = []
for i in android_english:
    if i[7] == '0':
        android_final.append(i)
for x in ios_english:
    if x[4] == '0.0':
        ios_final.append(x)
print(len(android_final))
print(len(ios_final))

8862
3222


## Most common app by Genre

We need to find app profiles that are successful on both markets.<br>
We'll begin our analysis by creating frequency table <br>
Columns to consider: <br>
```
ios: prime_genre <br>
android:Category , Genres
```

In [55]:
a = {'f':1, 's':2}
list(a.keys())

['f', 's']

Creating frquency table

In [57]:
def freq_table(dataset, index):
    table = {}
    total = 0
    for i in dataset:
        total+=1
        if i[index] in table:
            table[i[index]]+=1
        else:
            table[i[index]] = 1
            
    table_perc = {}
    for x in table:
        percentage = (table[x] / total) * 100
        table_perc[x] = percentage
    return table_perc
    

In [58]:
# First we are creating a freq table, but since later we need the % , we are initiating a total value variable
# Creating a seperate dict to store this percentage wise frequency table

In [59]:
def display_table(dataset, index):
    d_table = freq_table(dataset, index)
    table_list = []
    for i in d_table:
        new_tuple = (d_table[i], i)
        table_list.append(new_tuple)
    sorted_list = sorted(table_list, reverse = True)
    for val in sorted_list:
        print(val[1] , ':' , val[0])

In [60]:
# We need to finally show the data order wise descending .
# Soring in dictionary is difficult , so here we've created a new list of tuples

Examining frquency tables

In [62]:
display_table(ios_final, -5)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


We can see that among free ios english apps, Games genre is having the most number of apps, followed by entertainment, Photos & Videos.
General takeaway is ios appstore is dominated by apps which is designed for fun. Apps with practical purposes are less in number

In [64]:
%%time
display_table(android_final, 1)

FAMILY : 18.934777702550214
GAME : 9.693071541412774
TOOLS : 8.451816745655607
BUSINESS : 4.5926427443015125
LIFESTYLE : 3.9043105393816293
PRODUCTIVITY : 3.8930264048747465
FINANCE : 3.7011961182577298
MEDICAL : 3.5206499661475967
SPORTS : 3.39652448657188
PERSONALIZATION : 3.3175355450236967
COMMUNICATION : 3.238546603475513
HEALTH_AND_FITNESS : 3.080568720379147
PHOTOGRAPHY : 2.945159106296547
NEWS_AND_MAGAZINES : 2.798465357707064
SOCIAL : 2.663055743624464
TRAVEL_AND_LOCAL : 2.335815842924848
SHOPPING : 2.2455427668697814
BOOKS_AND_REFERENCE : 2.143985556307831
DATING : 1.8618821936357481
VIDEO_PLAYERS : 1.7941773865944481
MAPS_AND_NAVIGATION : 1.399232678853532
FOOD_AND_DRINK : 1.2412547957571656
EDUCATION : 1.1735499887158656
ENTERTAINMENT : 0.9591514330850823
LIBRARIES_AND_DEMO : 0.9365831640713158
AUTO_AND_VEHICLES : 0.9252990295644324
HOUSE_AND_HOME : 0.8237418190024826
WEATHER : 0.8011735499887158
EVENTS : 0.7109004739336493
PARENTING : 0.6544798013992327
ART_AND_DESIGN : 0.

Here in android playstore , a high number of apps are present for practical purposes unlike ios appstore 

In [66]:
display_table(android_final, -4)

Tools : 8.440532611148726
Entertainment : 6.070864364703228
Education : 5.348679756262695
Business : 4.5926427443015125
Productivity : 3.8930264048747465
Lifestyle : 3.8930264048747465
Finance : 3.7011961182577298
Medical : 3.5206499661475967
Sports : 3.4642292936131795
Personalization : 3.3175355450236967
Communication : 3.238546603475513
Action : 3.1031369893929135
Health & Fitness : 3.080568720379147
Photography : 2.945159106296547
News & Magazines : 2.798465357707064
Social : 2.663055743624464
Travel & Local : 2.324531708417964
Shopping : 2.2455427668697814
Books & Reference : 2.143985556307831
Simulation : 2.0424283457458814
Dating : 1.8618821936357481
Arcade : 1.8505980591288649
Video Players & Editors : 1.7716091175806816
Casual : 1.7490408485669149
Maps & Navigation : 1.399232678853532
Food & Drink : 1.2412547957571656
Puzzle : 1.128413450688332
Racing : 0.9930038366057323
Role Playing : 0.9365831640713158
Libraries & Demo : 0.9365831640713158
Auto & Vehicles : 0.92529902956443

Genre is more detailed, granular <br>
For this project to look from a bigger picture pov , we'll continue with Category

So, for ios , app designed for fun dominates number wise, whereas playstore still shows a balance

Now, we'd like to determine the kind of apps with the most users. <br>
One way to find out what genres are the most popular (have the most users) is to calculate the average <u>number of installs</u> for each app genre. For the Google Play data set, we can find this information in the <b>Installs</b> column, but this information is missing for the App Store data set. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the <b>rating_count_tot</b> column

Calculating the average number of user ratings per app genre on the App Store

## Most Popular app on AppStore by Genre

In [72]:
dict = freq_table(ios_final, -5)
free_list = []
for genre in dict:
    total = 0
    rating = 0
    for i in ios_final:
        if i[-5] == genre:
            total+=1
            f = float(i[5])
            rating+=f
    avg = rating/total
    new_tuple = (avg,genre)
    free_list.append(new_tuple)
    #print(genre, ':', avg) 

f = sorted(free_list, reverse = True)
for z in f:
    print(z[1] ,':', z[0])

Navigation : 86090.33333333333
Reference : 74942.11111111111
Social Networking : 71548.34905660378
Music : 57326.530303030304
Weather : 52279.892857142855
Book : 39758.5
Food & Drink : 33333.92307692308
Finance : 31467.944444444445
Photo & Video : 28441.54375
Travel : 28243.8
Shopping : 26919.690476190477
Health & Fitness : 23298.015384615384
Sports : 23008.898550724636
Games : 22788.6696905016
News : 21248.023255813954
Productivity : 21028.410714285714
Utilities : 18684.456790123455
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Business : 7491.117647058823
Education : 7003.983050847458
Catalogs : 4004.0
Medical : 612.0


In [73]:
# This has another easy way of calculation in o(n) complexity, the above code is having complexity o(n^2)

In [74]:
n_dict = {}
for i in ios_final:
    if i[-5] not in n_dict:
        n_dict[i[-5]] = [0,0]
    
    n_dict[i[-5]][0]+= float(i[5])
    n_dict[i[-5]][1]+=1
    
f_list = []
for genre, (rating,total) in n_dict.items():
    avg=rating/total
    f_list.append((avg,genre))
t_tuple = sorted(f_list, reverse = True)
for x in t_tuple:
    print(x[1], ':' , x[0])


Navigation : 86090.33333333333
Reference : 74942.11111111111
Social Networking : 71548.34905660378
Music : 57326.530303030304
Weather : 52279.892857142855
Book : 39758.5
Food & Drink : 33333.92307692308
Finance : 31467.944444444445
Photo & Video : 28441.54375
Travel : 28243.8
Shopping : 26919.690476190477
Health & Fitness : 23298.015384615384
Sports : 23008.898550724636
Games : 22788.6696905016
News : 21248.023255813954
Productivity : 21028.410714285714
Utilities : 18684.456790123455
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Business : 7491.117647058823
Education : 7003.983050847458
Catalogs : 4004.0
Medical : 612.0


In [75]:
print(n_dict)

{'Social Networking': [7584125.0, 106], 'Photo & Video': [4550647.0, 160], 'Games': [42705967.0, 1874], 'Music': [3783551.0, 66], 'Reference': [1348958.0, 18], 'Health & Fitness': [1514371.0, 65], 'Weather': [1463837.0, 28], 'Utilities': [1513441.0, 81], 'Travel': [1129752.0, 40], 'Shopping': [2261254.0, 84], 'News': [913665.0, 43], 'Navigation': [516542.0, 6], 'Lifestyle': [840774.0, 51], 'Entertainment': [3563577.0, 254], 'Food & Drink': [866682.0, 26], 'Sports': [1587614.0, 69], 'Book': [556619.0, 14], 'Finance': [1132846.0, 36], 'Education': [826470.0, 118], 'Productivity': [1177591.0, 56], 'Business': [127349.0, 17], 'Catalogs': [16016.0, 4], 'Medical': [3672.0, 6]}


On an average , Navigation has highest number of user reviews. 
However, if we further explore this category ,we can see Majority of reviews are covered by Google maps, waze - some big players in this segment

In [77]:
for i in ios_final:
    if i[-5] == 'Navigation':
        print(i[1],':', i[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


If we do the same analysis for Social Networking: where the average number is heavily influenced by a few giants like Facebook, Pinterest, Skype, etc.

In [79]:
for i in ios_final:
    if i[-5] == 'Social Networking':
        print(i[1],':', i[5])

Facebook : 2974676
Pinterest : 1061624
Skype for iPhone : 373519
Messenger : 351466
Tumblr : 334293
WhatsApp Messenger : 287589
Kik : 260965
ooVoo – Free Video Call, Text and Voice : 177501
TextNow - Unlimited Text + Calls : 164963
Viber Messenger – Text & Call : 164249
Followers - Social Analytics For Instagram : 112778
MeetMe - Chat and Meet New People : 97072
We Heart It - Fashion, wallpapers, quotes, tattoos : 90414
InsTrack for Instagram - Analytics Plus More : 85535
Tango - Free Video Call, Voice and Chat : 75412
LinkedIn : 71856
Match™ - #1 Dating App. : 60659
Skype for iPad : 60163
POF - Best Dating App for Conversations : 52642
Timehop : 49510
Find My Family, Friends & iPhone - Life360 Locator : 43877
Whisper - Share, Express, Meet : 39819
Hangouts : 36404
LINE PLAY - Your Avatar World : 34677
WeChat : 34584
Badoo - Meet New People, Chat, Socialize. : 34428
Followers + for Instagram - Follower Analytics : 28633
GroupMe : 28260
Marco Polo Video Walkie Talkie : 27662
Miitomo : 2

In [80]:
for i in ios_final:
    if i[-5] == 'Book':
        print(i[1],':', i[5])

Kindle – Read eBooks, Magazines & Textbooks : 252076
Audible – audio books, original series & podcasts : 105274
Color Therapy Adult Coloring Book for Adults : 84062
OverDrive – Library eBooks and Audiobooks : 65450
HOOKED - Chat Stories : 47829
BookShout: Read eBooks & Track Your Reading Goals : 879
Dr. Seuss Treasury — 50 best kids books : 451
Green Riding Hood : 392
Weirdwood Manor : 197
MangaZERO - comic reader : 9
ikouhoushi : 0
MangaTiara - love comic reader : 0
謎解き : 0
謎解き2016 : 0


For this niche, it shows some potential. One thing which can be done is take anyr popular book and turn it into an app where we could add different features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes about the book, or any engaging content. On top of that, we could also embed a dictionary within the app, so users don't need to exit our app to look up words in an external app.

This idea seems to fit well with the fact that the App Store is dominated by for-fun apps. This suggests the market might be a bit saturated with for-fun apps, which means a practical app might have more of a chance to stand out among the huge number of apps on the App Store.

For other popular genres: <br>
Weather apps — people generally don't spend too much time in-app, and the chances of making profit from in-app adds are low. Also, getting reliable live weather data may require us to connect our apps to non-free APIs.

Food and drink — examples here include Starbucks, Dunkin' Donuts, McDonald's, etc. So making a popular food and drink app requires actual cooking and a delivery service, which is outside the scope of this analysis/project.

Finance apps — these apps involve banking, paying bills, money transfer, etc. Building a finance app requires domain knowledge, and we don't want to hire a finance expert just to build an app.

## Google play dataset analysis


In [84]:
g_play = {}
g_list = []
for i in android_final:
    
    cat = i[1]
    inst = i[5]
    if cat not in g_play:
        g_play[cat] = [0,0]
    inst = inst.replace(',' , '')
    inst = inst.replace('+' , '')
    g_play[cat][0]+=int(inst)
    g_play[cat][1]+=1
for category, (total_install, total_apps) in g_play.items():
    avg = total_install/total_apps
    g_list.append([avg, category])

new_list = sorted(g_list, reverse =True)
for i in new_list:
    print(i[1], ':' , i[0])

COMMUNICATION : 38456119.167247385
VIDEO_PLAYERS : 24727872.452830188
SOCIAL : 23253652.127118643
PHOTOGRAPHY : 17805627.643678162
PRODUCTIVITY : 16787331.344927534
GAME : 15560965.599534342
TRAVEL_AND_LOCAL : 13984077.710144928
ENTERTAINMENT : 11640705.88235294
TOOLS : 10682301.033377837
NEWS_AND_MAGAZINES : 9549178.467741935
BOOKS_AND_REFERENCE : 8767811.894736841
SHOPPING : 7036877.311557789
PERSONALIZATION : 5201482.6122448975
WEATHER : 5074486.197183099
HEALTH_AND_FITNESS : 4188821.9853479853
MAPS_AND_NAVIGATION : 4056941.7741935486
FAMILY : 3694276.334922527
SPORTS : 3638640.1428571427
ART_AND_DESIGN : 1986335.0877192982
FOOD_AND_DRINK : 1924897.7363636363
EDUCATION : 1820673.076923077
BUSINESS : 1712290.1474201474
LIFESTYLE : 1437816.2687861272
FINANCE : 1387692.475609756
HOUSE_AND_HOME : 1331540.5616438356
DATING : 854028.8303030303
COMICS : 817657.2727272727
AUTO_AND_VEHICLES : 647317.8170731707
LIBRARIES_AND_DEMO : 638503.734939759
PARENTING : 542603.6206896552
BEAUTY : 51315

On average, communication apps have the most installs: 38,456,119. This number is heavily skewed up by a few apps that have over one billion installs (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts), and a few others with over 100 and 500 million installs:

In [86]:
for i in android_final:
    if i[1] == 'COMMUNICATION':
        print(i[0], ':' , i[5])

WhatsApp Messenger : 1,000,000,000+
Messenger for SMS : 10,000,000+
My Tele2 : 5,000,000+
imo beta free calls and text : 100,000,000+
Contacts : 50,000,000+
Call Free – Free Call : 5,000,000+
Web Browser & Explorer : 5,000,000+
Browser 4G : 10,000,000+
MegaFon Dashboard : 10,000,000+
ZenUI Dialer & Contacts : 10,000,000+
Cricket Visual Voicemail : 10,000,000+
TracFone My Account : 1,000,000+
Xperia Link™ : 10,000,000+
TouchPal Keyboard - Fun Emoji & Android Keyboard : 10,000,000+
Skype Lite - Free Video Call & Chat : 5,000,000+
My magenta : 1,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Seznam.cz : 1,000,000+
Antillean Gold Telegram (original version) : 100,000+
AT&T Visual Voicemail : 10,000,000+
GMX Mail : 10,000,000+
Omlet Chat : 10,000,000+
My Vodacom SA : 5,000,000+
Microsoft Edge : 5,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Calls & Text by Mo+ : 5,000,000+
free 

If we remove all the communication apps , that have over 100 Million downloads , average will be reduced by 10 times(approx)

In [88]:
ideal_avg = []
for i in android_final:
    download = i[5]
    download = download.replace('+','')
    download = download.replace(',','')
    if(i[1] == 'COMMUNICATION' and float(download) <100000000):
        ideal_avg.append(float(download))
print( sum(ideal_avg) / len(ideal_avg))

3603485.3884615386


If we look into the Video Player category, which is the runner-up with 24727872 installs, the pattern is same. The market share is dominated by apps like Youtube, Google Play Movies & TV, or MX Player. <br>
The pattern is repeated for social apps (where we have giants like Facebook, Instagram, Google+, etc.), photography apps (Google Photos and other popular photo editors), or productivity apps (Microsoft Word, Dropbox, Google Calendar, Evernote, etc.).

These app genres are more popular than they really are. Moreover, these niches seem to be dominated by a few large players who are hard to compete against.

The game genre seems popular, but previously we found out this part of the market seems a bit saturated, so we'd like to come up with a different app recommendation if possible.

The books and reference genre looks fairly popular as well, with an average number of installs of 8767811. It's interesting to explore this in more depth, since we found this genre has some potential to work well on the App Store, and our aim is to recommend an app genre that shows potential for being profitable on both the App Store and Google Play.

In [90]:
for i in android_final:
    if i[1] == 'VIDEO_PLAYERS':
        print(i[0], ':' , i[5])

YouTube : 1,000,000,000+
All Video Downloader 2018 : 1,000,000+
Video Downloader : 10,000,000+
HD Video Player : 1,000,000+
Iqiyi (for tablet) : 1,000,000+
Video Player All Format : 10,000,000+
Motorola Gallery : 100,000,000+
Free TV series : 100,000+
Video Player All Format for Android : 500,000+
VLC for Android : 100,000,000+
Code : 10,000,000+
Vote for : 50,000,000+
XX HD Video downloader-Free Video Downloader : 1,000,000+
OBJECTIVE : 1,000,000+
Music - Mp3 Player : 10,000,000+
HD Movie Video Player : 1,000,000+
YouCut - Video Editor & Video Maker, No Watermark : 5,000,000+
Video Editor,Crop Video,Movie Video,Music,Effects : 1,000,000+
YouTube Studio : 10,000,000+
video player for android : 10,000,000+
Vigo Video : 50,000,000+
Google Play Movies & TV : 1,000,000,000+
HTC Service － DLNA : 10,000,000+
VPlayer : 1,000,000+
MiniMovie - Free Video and Slideshow Editor : 50,000,000+
Samsung Video Library : 50,000,000+
OnePlus Gallery : 1,000,000+
LIKE – Magic Video Maker & Community : 50,

In [91]:
for i in android_final:
    if i[1] == 'SOCIAL':
        print(i[0], ':' , i[5])

Facebook : 1,000,000,000+
Facebook Lite : 500,000,000+
Tumblr : 100,000,000+
Social network all in one 2018 : 100,000+
Pinterest : 100,000,000+
TextNow - free text + calls : 10,000,000+
Google+ : 1,000,000,000+
The Messenger App : 1,000,000+
Messenger Pro : 1,000,000+
Free Messages, Video, Chat,Text for Messenger Plus : 1,000,000+
Telegram X : 5,000,000+
The Video Messenger App : 100,000+
Jodel - The Hyperlocal App : 1,000,000+
Hide Something - Photo, Video : 5,000,000+
Love Sticker : 1,000,000+
Web Browser & Fast Explorer : 5,000,000+
LiveMe - Video chat, new friends, and make money : 10,000,000+
VidStatus app - Status Videos & Status Downloader : 5,000,000+
Love Images : 1,000,000+
Web Browser ( Fast & Secure Web Explorer) : 500,000+
SPARK - Live random video chat & meet new people : 5,000,000+
Golden telegram : 50,000+
Facebook Local : 1,000,000+
Meet – Talk to Strangers Using Random Video Chat : 5,000,000+
MobilePatrol Public Safety App : 1,000,000+
💘 WhatsLov: Smileys of love, sti

In [92]:
for i in android_final:
    if i[1] == 'BOOKS_AND_REFERENCE':
        print(i[0], ':' , i[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

Sorted

In [94]:
book_list = []
for i in android_final:
    if i[1] == 'BOOKS_AND_REFERENCE':
        cat = i[0]
        download = i[5]
        download = download.replace('+','')
        download = download.replace(',','')
        book_list.append((int(download),cat))
new_list = sorted(book_list, reverse = True)
#for i in new_list:
#    print(i[0], ':', i[1])
for download,cat in new_list:
    for x in android_final:
        if x[0] == cat:
            print(cat, '', x[5])

Google Play Books  1,000,000,000+
Wattpad 📖 Free Books  100,000,000+
Bible  100,000,000+
Audiobooks from Audible  100,000,000+
Amazon Kindle  100,000,000+
Wikipedia  10,000,000+
Spanish English Translator  10,000,000+
Quran for Android  10,000,000+
Oxford Dictionary of English : Free  10,000,000+
NOOK: Read eBooks & Magazines  10,000,000+
Moon+ Reader  10,000,000+
JW Library  10,000,000+
HTC Help  10,000,000+
FBReader: Favorite Book Reader  10,000,000+
English Hindi Dictionary  10,000,000+
English Dictionary - Offline  10,000,000+
Dictionary.com: Find Definitions for English Words  10,000,000+
Dictionary - Merriam-Webster  10,000,000+
Dictionary  10,000,000+
Cool Reader  10,000,000+
Aldiko Book Reader  10,000,000+
Al-Quran (Free)  10,000,000+
Al'Quran Bahasa Indonesia  10,000,000+
Al Quran Indonesia  10,000,000+
Read books online  5,000,000+
English to Hindi Dictionary  5,000,000+
Ebook Reader  5,000,000+
Dictionary - WordWeb  5,000,000+
Bible KJV  5,000,000+
Ancestry  5,000,000+
AlRea

Here also we can see , the average is skewed because of small numbers of heavily installed apps

## Conclusions

In this project, we explored data from both the App Store and Google Play to understand what kinds of mobile apps tend to perform well. Our goal was to identify an app idea that could succeed on both platforms.

Based on the analysis, one promising direction is to build an app around a popular or recently released book. Both markets already have plenty of basic e-book or library apps, so simply offering the book itself won’t stand out. However, combining the book with engaging added features could make it far more appealing.

For example, the app could include daily quotes, an audio version, interactive quizzes, or even a community space where readers discuss themes and insights. This kind of enhanced, experience-driven book app has the potential to perform well across both stores.